### DISCUSSION TITLE
advice on markers layout and possible vicon settings

### DISCUSSION THREAD

**Discussion by Hurisa** [USC-ACTLab/crazyswarm#303] [Q&A]
Hi everyone,

Is there any advice on the 4 markers layout (6.5mm) for the cf? I know it must be asymmetrical, but the fact is I can't get a stable point cloud. I'm using the mocap deck and attach a marker on one of the motor mounts similar to https://crazyswarm.readthedocs.io/en/latest/usage.html.
I have tried with a single larger marker (9.5mm) and then I could get stable tracking but the issue is that I want to send angular velocity commands to have the swarm flying autonomously and part of my algorithm is that each drone avoids its neighbours based on Pose.
Alternative is there particular properties for the vicon cameras that may help me?
I know this might be a bit vague but hopefully we can discuss some solution.

Thanks

**Comment by Hurisa** [comment]
By the way I was able to improve detection by adjusting the position of some of the cameras to better cover the gorund plane.
However, when I turn on the cf I get an unstable point cloud again... I guess it might be the lights?!
Is there any parameter I can tune to improve this behaviour?

I also get these messages
```
[pcl::IterativeClosestPoint::computeTransformation] Not enough correspondences found. Relax your threshold parameters.
[ WARN] [1604080375.603129017]: ICP did not converge! for object cf3
[ WARN] [1604080375.603225679]: No updated pose for CF cf3 for 0.009912 s.

```
Any help is much appreciated

**Comment by whoenig** [comment]
* In the camera view, make sure that you markers are visible (at least 3-5 pixels per marker in as many cameras as possible).
* In the 3D perspective view, you can select individual markers and lines are drawn that show you which camera can see that marker. Make sure that at least 4 cameras can see each marker in the desired flight volume. Adjusting camera position or angles can help to improve the coverage for your use case.
* For the 'unstable point cloud' check the raw camera view. You can also try reducing the IR strobe intensity, in case there are unwanted reflections. When you calibrate, make sure you are using the masking feature to avoid any spurious background detection.

**Comment by jpreiss** [comment]
Check that the frame rate of your mocap system is not too high. Someone set our Vicon to 400Hz and it was causing all kinds of problems. Even 10 or 20 Hz should work fine.

If you have few CFs and marker decks, you could make a unique layout for each CF and use your mocap built in object tracking. It is generally more robust.

**Comment by Hurisa** [comment]
Thank you @jpreiss and @whoenig  so much for your advice.
I'll let you know if I make it! Fortunately I have access to my lab over the weekend.
